subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/TestRunnerFactory.kt

```diff
@@ -15,35 +11,30 @@ import com.avito.runner.scheduler.runner.TestRunnerExecutionState
 import com.avito.runner.scheduler.runner.TestRunnerImpl
 import com.avito.runner.scheduler.runner.scheduler.TestExecutionScheduler
 import com.avito.runner.service.DeviceWorkerPool
-import com.avito.runner.service.DeviceWorkerPoolImpl
-import com.avito.runner.service.listener.CompositeListener
-import com.avito.runner.service.listener.TestListener
+import com.avito.runner.service.DeviceWorkerPoolProvider
 import com.avito.runner.service.worker.device.Device
-import com.avito.time.DefaultTimeProvider
-import com.avito.time.TimeProvider
 import kotlinx.coroutines.channels.ReceiveChannel
 import java.io.File
 
 internal class TestRunnerFactory(
-    private val config: TestRunnerFactoryConfig
+    outputDir: File,
+    private val config: TestRunnerFactoryConfig,
+    private val deviceWorkerPoolProvider: DeviceWorkerPoolProvider,
+    private val testMetricsListener: TestMetricsListener
 ) {
     private val loggerFactory = config.loggerFactory
-    private val timeProvider: TimeProvider = DefaultTimeProvider()
-    private val statsDSender: StatsDSender = StatsDSender.Impl(
-        config = config.metricsConfig.statsDConfig,
-        loggerFactory = loggerFactory
-    )
-    private val testMetricsSender = TestMetricsListenerImpl(
-        testMetricsSender = TestMetricsSender(
-            statsDSender = statsDSender,
-            prefix = config.metricsConfig.runnerPrefix
-        ),
-        timeProvider = timeProvider,
-        loggerFactory = config.loggerFactory
-    )
+
+    /**
+     * i.e {projectDir}/output/test-runner/
+     */
+    private val testRunnerOutputDir: File by lazy {
```

Comment:
Please take a look at: https://github.com/avito-tech/avito-android/blob/68d514f0eb4e56c244946151d2f41716c4162e8d/subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt#L129
Does it clashes somehow? Maybe move more logic from gradle plugin there?
On plugin side it's not clear what outputDir user should specify

Reply:
As I understand we have root output-dir and some sub-dirs. In that case, Factory need a `root output dir` to create a sub-dir.
So in the plugin, we could rename the property as `pluginRootOutputDir`